SalesforceCommerceCloud/pwa-kit
# [Localization] Add localized urls to the current page's html

# Description

For search engine crawlers, we need to let them know of the localized variations of the current page, via`hreflang` links that would go into the html head section.

GUS ticket: https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07AH000000OdIXYA0/view
For more details on how to use `hreflang` links: https://developers.google.com/search/docs/advanced/crawling/localized-versions

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Added `hreflang` links for all of the supported locales
- Added 2 fallback links: one for the general `en` locale, while the other one is for all other user locales that the app does not support
- Tweaked `buildUrlLocale()` util function

# How to Test-Drive This PR

1. Go to homepage 
2. Inspect the html's `<head>` and look for the `hreflang` links
3. Verify that there are ones for:
- all of the supported locales
- fallback general locale `en`
- fallback locale (`x-default`) for all user locales that the app does not support
4. Switch the current locale to Japanese for example
5. The same set of `hreflang` links should still be in the html head. But they might be scattered throughout due to re-rendering.
6. Navigate to another page
7. Verify the `hreflang` links again

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
